File: brickport-services/src/queries/player-queries.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace BrickPort.Services.Queries
{
    public class Player 
    {
        public string PlayerId { get; set; }
        public string PlayerName { get; set; }
    }
    public interface IPlayerQueries
    {
        Task<IEnumerable<Player>> GetAsync();
        Task<Player> GetAsync(string id);
    }
}

File: brickport-ui/src/infrastructure/brickport-services/brickport-services-api/queries/player-queries.ts
import { IPlayerQueries, IPlayer } from "../../../../brickport-services/queries/player-queries";
import { BrickportApiClient } from "../api-client";

export class ApiPlayerQueries implements IPlayerQueries {
    
    private readonly _apiClient: BrickportApiClient;
    
    constructor(endpoint: string) {
        this._apiClient = new BrickportApiClient(endpoint, true);
    }

    getAsync(): Promise<IPlayer[]>;
    getAsync(id: string): Promise<IPlayer>;
    getAsync(id?: any): Promise<IPlayer | IPlayer[]> {
        if (id == null) {
            return this._apiClient.get(`/${id}`);
        }
        return this._apiClient.get();
    }
    
}